test/jdk/java/foreign/TestMemoryAlignment.java

```diff
@@ -124,10 +132,68 @@ public void testPackedAccess() {
         }
     }
 
+    @Test(dataProvider = "alignments")
+    public void testActualByteAlignment(long align) {
+        if (align > (1L << 10)) {
+            return;
+        }
+        try (Arena arena = Arena.ofConfined()) {
+            var segment = arena.allocate(4, align);
+            assertTrue(segment.maxByteAlignment() >= align);
+            // Even power of two?
+            assertEquals(Long.bitCount(segment.maxByteAlignment()), 1);
+            assertEquals(segment.asSlice(1).maxByteAlignment(), 1);
+        }
+    }
+
+    public void testActualByteAlignmentMappedSegment() throws IOException {
+        File tmp = File.createTempFile("tmp", "txt");
+        try (FileChannel channel = FileChannel.open(tmp.toPath(), StandardOpenOption.READ, StandardOpenOption.WRITE);
+             Arena arena = Arena.ofConfined()) {
+            var segment =channel.map(FileChannel.MapMode.READ_WRITE, 0L, 32L, arena);
+            assertTrue(segment.maxByteAlignment() >= Long.BYTES);
```

Comment:
Is this always the case? Smells of platform-dependent...

Reply:
I think the alignment is always the biggest of the directly supported primitive types so maybe we should change to `Integer.BYTES` here.

Reply:
Yes, but, I mean, we don't have a test for checking what's the alignment of `malloc` - so I wonder why we should have a test for mapped segments - it's not like the API makes any promises there, right?